### PR TITLE
CRM-21687 Fix upgrade issue on MariaDB for current_timestamp

### PR DESCRIPTION
**Pull request by gah242s**
https://issues.civicrm.org/jira/browse/CRM-21687

Overview
----------------------------------------
_A brief description of the pull request. Try to keep it non-technical._

Before
----------------------------------------
_The current status. Please provide screenshots or gifs ([LICEcap](http://www.cockos.com/licecap/), [SilentCast](https://github.com/colinkeenan/silentcast)) where appropriate._

After
----------------------------------------
_What has been changed. Please provide screenshots or gifs ([LICEcap](http://www.cockos.com/licecap/), [SilentCast](https://github.com/colinkeenan/silentcast)) where appropriate._

Technical Details
----------------------------------------
_If the PR introduces noteworthy technical changes, please describe them here. Provide code snippets if necessary_

Comments
----------------------------------------
_Anything else you would like the reviewer to note_

### CODE REVIEW

**Comment by mlutfy** [comment]
jenkins, test this please

File: CRM/Logging/Schema.php [hunk 1]
```diff
@@ -487,7 +487,7 @@ public function fixSchemaDifferencesForAll($rebuildTrigger = FALSE) {
   public function fixTimeStampAndNotNullSQL($query) {
     $query = str_ireplace("TIMESTAMP NOT NULL", "TIMESTAMP NULL", $query);
     $query = str_ireplace("DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP", '', $query);
```
Comment (mlutfy) [inline CRM/Logging/Schema.php]: Should this `CURRENT_TIMESTAMP` be changed as well?

Comment (gah242s) [reply CRM/Logging/Schema.php]: According to the way I read the code, I think the second instance should take care of anything left behind from the first.  Perhaps it's safer, especially in the long run, to do both?

**Comment by seamuslee001** [comment]
@gah242s what version of MariaDB are you running?. Had you run the dockerwhen cleanup? i take it this was on just the log tables right that it was failing?

**Comment by gah242s** [comment]
@seamuslee001 MariaDB 10.2 from CentOS 7 SCL.  There's no Docker involved, so I'm assuming that no I didn't and that it wasn't needed.  Best I can tell, it was only the log table.  After successfully installing the update, I'm seeing another error now of Unknown column 'a.cancel_button_text' in 'field list', but I believe that it is unrelated to this PR, though it may be MariaDB related.

**Comment by seamuslee001** [comment]
@gah242s that 2nd error is suggestive that you haven't run the db upgrade script at `/civicrm/upgrade?reset=1`

**Comment by gah242s** [comment]
@seamuslee001 It said that it ran and completed.  When I attempt to rerun, I get:

 Sorry, due to an error, we are unable to fulfill your request at the moment. You may want to contact your administrator or service provider with more details about what action you were performing when this occurred.
Your database has already been upgraded to CiviCRM 4.7.29

This was after rolling back, applying the code fix, and re-running the upgrade script.

**Comment by seamuslee001** [comment]
hmmm ok we will leave that 2nd error maybe an issue to do with logging.

**Comment by seamuslee001** [comment]
Jenkins add to whitelist

**Comment by seamuslee001** [comment]
Jenkins re test this please

**Comment by eileenmcnaughton** [comment]
closed in favour of #11580

**Pull request closed without merging**